nemo/collections/nlp/parts/nlp_overrides.py

```diff
@@ -64,17 +82,55 @@
         optim_state_to_sharding_state,
     )
     from megatron.core.transformer.module import Float16Module as MCoreFloat16Module
+    from megatron.core.transformer.transformer_layer import TransformerLayer as MCoreTransformerLayer
```

Comment:
@erhoo82 @ericharper do we want to move this to a separate try except block ? Cause we would have `HAVE_MEGATRON_CORE=False` if the import fails with `TransformerEngine not found error`, although megatron core would be available and can cause error for blocks that depend just on mcore and not TE.

Reply:
We can leave it as is for now. There is a separate  effort to improve the import guarding.